CMakeLists.txt

```
@@ -1,7 +1,11 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
 
-cmake_minimum_required(VERSION 3.9)
+if(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.30")
+  cmake_minimum_required(VERSION 3.10)
+else()
+  cmake_minimum_required(VERSION 3.9)
+endif()
```

```suggestion
cmake_minimum_required(VERSION 3.10)
```

Could we make this not conditional, and just require 3.10 ?

---

Also, the policy below:

```
cmake_policy(SET CMP0057 NEW)
```

is outdated in my understanding, because it will always to NEW with >= 3.3

Should that part be removed ?

---

According to <https://cmake.org/cmake/help/latest/policy/CMP0057.html>, cmake >= 3.3 just add this policy but the default behavior is OLD.So I think `cmake_policy(SET CMP0057 NEW)` is still needed here.

---

I also add some policy settings to use cmake a more modern way.Could you please review it to check if it's OK?

---

There is something I do not understand here.

CMake policy CMP0054 was added in 3.1

The doc https://cmake.org/cmake/help/latest/command/cmake_minimum_required.html#command:cmake_minimum_required for CMAKE_MINIMUM_REQUIRED says:

> All policies known to the running version of CMake and introduced in the min (or max, if specified) version or earlier will be set to use NEW behavior.

So, `cmake_minimum_required(VERSION 3.10)` should set policy CMP0054 to NEW, because that policy was added in 3.1, and 3.1 <= 3.10.

Why the need to set CMP0057 explicitly ?

Likewise for CMP0057 (3.3), CMP0067 (3.8)

---

Thanks, I missed this message before. These policies are cleaned now.